modules/apps/object/object-web/types/src/main/resources/META-INF/resources/js/components/ObjectDetails/EditObjectDetails.d.ts

```diff
@@ -12,18 +12,19 @@
  * details.
  */
 
-/// <reference types="react" />
-
+import {FormError} from '@liferay/object-js-components-web';
+import React from 'react';
 import './ObjectDetails.scss';
 export declare type KeyValuePair = {
 	key: string;
 	value: string;
 };
 interface EditObjectDetailsProps {
```

Comment:
Since we are going to rename the hook that is declaring this value and setValue, I don't think it will be necessary to add Generic Types here. Generic Types are used in the situation of more than 1 call of a component or functions and these different calls will use different data structures to do the same thing inside this component, in this case we will call this component only once and using the same data structure as the ObjectDefiniton so it's not needed, see more here: https://devtrium.com/posts/react-typescript-using-generics-in-react#why-generics-can-be-very-useful